packages/core/src/util/file-type.ts

```diff
@@ -92,13 +98,13 @@ export function isLikelyTextFile(fileBuffer: Buffer): boolean {
  * matches it against known file type headers.
  *
  * @param fileBuffer - The `fileBuffer` parameter is a `Buffer` object that represents the
- * contents of a file. The first 5 bytes of this buffer, which generally contain the magic number, are used to
+ * contents of a file. The first 6 bytes of this buffer, which generally contain the magic number, are used to
  * identify the file type.
  * @returns The function `detectFileType` returns a string representing the detected file type.
  */
-export function detectFileType(document: string): [string, string] {
+export function detectFileType(document: Buffer): { mimeType: string; extension: string } {
   const maxBytesNeeded = 6; //NOTE: if you update detectFileType, you might need to update this number
-  const fileBuffer = Buffer.from(document.slice(0, maxBytesNeeded));
+  const fileBuffer = document.slice(0, maxBytesNeeded);
```

Comment:
the logic of Bufer.from is now just used before calling this function. This function is now shared by CW and CQ.